website/static/js/fangorn.js

```diff
@@ -88,31 +89,70 @@ function findByTempID(parent, tmpID) {
     return item;
 }
 
-function cancelUploads (row) {
+/**
+ * Cancel pending upload(s)
+ * @this Treebeard.controller
+ * @param {Object|null} row Treebeard row containing the file to cancel. If omitted, cancel all pending uploads.
+ */
+function cancelUploads(row) {
     var tb = this;
-    var uploading = tb.dropzone.getUploadingFiles();
-    var rejected = tb.dropzone.getRejectedFiles();
-    var queued = tb.dropzone.getQueuedFiles();
-    var filesArr = uploading.concat(queued, rejected);
-
-    for (var i = 0; i < filesArr.length; i++) {
-        var j = filesArr[i];
-        if(!row) {
-            var parent = j.treebeardParent || tb.dropzoneItemCache;
-            var item = findByTempID(parent, j.tmpID);
-            tb.dropzone.removeFile(j);
-            tb.deleteNode(parent.id,item.id);
-        } else {
-            tb.deleteNode(row.parentID,row.id);
-            if(row.data.tmpID === j.tmpID){
-                tb.dropzone.removeFile(j);
+    var cancelableStatuses = [Dropzone.UPLOADING, Dropzone.QUEUED];
+    // Select files that are uploading, queued, or rejected (!accepted)
+    var filesArr = tb.dropzone.files.filter(function(file) {
+        return cancelableStatuses.indexOf(file.status) > -1 || !file.accepted;
+    });
+    // Remove all queued files
+    if (row === undefined) {
```

Comment:
I do find where `cancelUploads` is [called](https://github.com/el1t/osf.io/blob/86896d674ba92087da624f33a4f8860651950a93/website/static/js/fangorn.js#L1823) without a `row` (though I haven't figured out how to get there in the UI).

👍